src/main/java/org/interledger/cryptoconditions/ConditionImpl.java

```diff
@@ -34,21 +46,63 @@ public ConditionImpl(ConditionType type, EnumSet<FeatureSuite> features, byte[]
 		this.maxFulfillmentLength = maxFulfillmentLength;
 	}
 
+	public ConditionImpl(String URI) {
+		if (URI == null)
+			throw new RuntimeException("serializedCondition == null");
+		if ("".equals(URI.trim()))
+			throw new RuntimeException("serializedCondition was an empy string");
+		if (!URI.startsWith("cc:"))
+			throw new RuntimeException("serializedCondition must start with 'cc:'");
+		
+		java.util.regex.Matcher m = p.matcher(URI);
+		if (!m.matches())
+			throw new RuntimeException(
+					"serializedCondition '" + URI + "' doesn't match " + ConditionImpl.CONDITION_REGEX);
+
+		String[] pieces = URI.split(":");
+		if (pieces.length != 5) 
+			throw new RuntimeException("The URI for the condition '"+URI+"' was expected to contain 5 fields separated by ':'");
+
+		String BASE16Type = pieces[1], BASE16FeatureBitMask = pieces[2], 
+				BASE64URLFingerprint = pieces[3], BASE10MaxFulfillmentLength = pieces[4];
+		
+		this.type = ConditionType.valueOf(Integer.parseInt(BASE16Type , 16));
+		this.features = FeatureSuite.bitMask2EnumSet(Integer.parseInt(BASE16FeatureBitMask, 16));
+		this.fingerprint = Base64Url.decode(BASE64URLFingerprint);
+		this.maxFulfillmentLength = Integer.parseInt(BASE10MaxFulfillmentLength);
+	}
+
+	@Override
 	public ConditionType getType() {
 		return this.type;
 	}
 	
-
+	@Override
 	public EnumSet<FeatureSuite> getFeatures(){
 		return this.features;
 	}
 	
 
+	@Override
 	public byte[] getFingerprint(){
 		return this.fingerprint;
 	}
 
+	@Override
 	public int getMaxFulfillmentLength() {
 		return this.maxFulfillmentLength;
 	}
+	
+	@Override
+	public String toURI() {
```

Comment:
We should put this logic in a ConditionWriter and have this as a convenience function for calling that logic.